hdf5/src/hl/references/legacy.rs

```diff
@@ -0,0 +1,69 @@
+//! The pre v1.12.0 reference types.
+
+use crate::internal_prelude::*;
+
+use hdf5_sys::{
+    h5o::H5O_type_t,
+    h5p::H5P_DEFAULT,
+    h5r::{hobj_ref_t, H5Rcreate, H5Rdereference, H5Rget_obj_type2},
+};
+use hdf5_types::H5Type;
+
+#[cfg(not(feature = "1.12.0"))]
+use hdf5_sys::h5r::H5R_OBJECT as H5R_OBJECT1;
+#[cfg(feature = "1.12.0")]
+use hdf5_sys::h5r::H5R_OBJECT1;
+
+use crate::{Location, ObjectReference};
+
+#[repr(transparent)]
+#[derive(Debug, Copy, Clone)]
+pub struct ObjectReference1 {
```

Comment:
Move to `hdf5-types`

Reply:
I have a version with this moved but if we can't move StdReference does it make sense to keep it where it is for consistency?

Reply:
I am OK keeping it here if we can't move `StdReference`